lib/auth/auth_test.go

```diff
@@ -1105,8 +1105,13 @@ func TestServer_AugmentContextUserCertificates(t *testing.T) {
 	const username = "llama"
 	const pass = "secret!!1!"
 
+	// Use a >1 list of principals.
+	// This is enough to cause ordering issues between the TLS and SSH principal
+	// lists, which caused a bug in the device trust preview.
+	principals := []string{"login0", "-teleport-internal-join", username}
```

Comment:
If we want to be accurate, I believe that `-teleport-internal-join` is _appended_ to the principals list for SSH certs.

Reply:
Done. +1 for being accurate (also the test still breaks without the fix).